### PR TITLE
georgettica/login to owner cluster

### PR DESCRIPTION
**Pull request by georgettica**
- feat(ocm-login-owner): login to owner cluster

### CODE REVIEW

**Comment by clcollins** [comment]
Can you be more descriptive what this is doing? I think I get it reading the code, but `login to owner cluster` isn't real clear, and I want to make sure I understand.

Regarding the mixed bash and python - I'm not going to say it's enough to downvote the PR, but would this be better to just use python for the whole thing rather than dragging it in at the end?

The code itself looks good, though!

File: utils/bin/get-shards-clusterid [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+parser.add_argument(
+    "-c",
+    "--cluster",
+    default=os.getenv("CLUSTER"),
+    help="""the cluster identifier (name/id/external) which you want to find it's shardid', can be set using the CLUSTER field """,
+)
+args = parser.parse_args()
+
+if not args.cluster:
+    sys.stderr.write(parser.print_usage())
```
Comment (T0MASD) [inline utils/bin/get-shards-clusterid:25]: I'm getting this error in ocm-container
```shell
usage: get-shards-clusterid [-h] [-c CLUSTER]
Traceback (most recent call last):
  File "/mnt/Development/runbooks/bla.py", line 26, in <module>
    sys.stderr.write(usage) #(parser.print_usage()) 
TypeError: write() argument must be str, not None   
```
I think it's fine to just 
```suggestion
    parser.print_usage()
```

**Review by T0MASD — COMMENTED** [review]
Overall looks good and I like this. Only one detail, if I'm not logged in I'm gettng an error. 
```
Traceback (most recent call last):
  File "/root/./get-shards-clusterid", line 35, in <module>
    described_clusters_raw = convert_to_dict(res)
  File "/root/./get-shards-clusterid", line 12, in convert_to_dict
    return json.loads(stringified_bytes)
  File "//usr/lib64/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "//usr/lib64/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "//usr/lib64/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
Maybe run `ocm config get url` or similar to check if logged in?

**Comment by georgettica** [comment]
Fixing your comment now @T0MASD

**Comment by georgettica** [comment]
@T0MASD can you look again?

I can add README aswell

File: utils/bin/get-shards-clusterid [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+# get_ocm_user calls `ocm whoami` and filters out the username from the response
+# to use in generating cluster names
+# copied from './create-cluster' as packages in python are problematic
+def get_ocm_user():
+    command = ["ocm", "whoami"]
+
+    try:
+        output = subprocess.run(command, capture_output=True, check=True)
+    except subprocess.CalledProcessError:
+        sys.stderr.write("Error retrieving user with `ocm whoami`; are you logged in?")
```
Comment (T0MASD) [inline utils/bin/get-shards-clusterid:28]:
```suggestion
        sys.stderr.write("Error retrieving user with `ocm whoami`; are you logged in?\n")
```

**Review by T0MASD — COMMENTED** [review]
Looking nice. I've tried running `./ocm-login-owner <cluster_id>` but I get
```
Error: exec: "xdg-open": executable file not found in $PATH
FAILURE: unable to login
```

I ran `dnf install xdg-utils` and re-ran, but then I get a different error:
```
/usr/bin/xdg-open: line 881: www-browser: command not found
/usr/bin/xdg-open: line 881: links2: command not found
/usr/bin/xdg-open: line 881: elinks: command not found
/usr/bin/xdg-open: line 881: links: command not found
/usr/bin/xdg-open: line 881: lynx: command not found
/usr/bin/xdg-open: line 881: w3m: command not found
```

**Comment by georgettica** [comment]
@T0MASD this PR is mostly to set the groundwork for `backplane`, I added to `ocm-login` as this is the easiest way to show how this works

The tool will be added to the downstream once merged

**Comment by T0MASD** [comment]
thanks
/lgtm

**Pull request merged**